### PR TITLE
K8SPXC-280 Add support for compressed backups

### PR DESCRIPTION
**Pull request by tplavcic**
[K8SPXC-280](https://jira.percona.com/browse/K8SPXC-280)

This is adding possibility to enable compressed backups in cr.
Corresponding PR for backup docker images is here: https://github.com/percona/percona-docker/pull/229

### CODE REVIEW

**Comment by it-percona** [comment]
[CLA assistant check](https://cla-assistant.percona.com:443/percona/percona-xtradb-cluster-operator?pullRequest=421) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/percona-xtradb-cluster-operator?pullRequest=421) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/percona-xtradb-cluster-operator?pullRequest=421) it.</sub>

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| scaling | passed |
| recreate | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| limits | passed |
| monitoring | passed |
| restore-to-encrypted-cluster | failed |
| upgrade | failed |
| monitoring-2-0 | passed |
| upgrade-consistency | passed |
| security-context | passed |
| demand-backup | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| scheduled-backup | failed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| big-data | passed |
| recreate | passed |
| restore-to-encrypted-cluster | failed |
| storage | failed |
| init-deploy | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| self-healing | passed |
| upgrade-consistency | passed |
| security-context | passed |
| self-healing-advanced | passed |
| demand-backup | passed |
| scheduled-backup | failed |
| operator-self-healing | passed |
| demand-backup-encrypted-with-tls | failed |

File: deploy/cr.yaml [hunk 1]
```diff
@@ -22,6 +22,10 @@ spec:
 #    livenessDelaySec: 300
 #    forceUnsafeBootstrap: false
 #    configuration: |
+#      [sst]
+#      xbstream-opts=--decompress
+#      [xtrabackup]
+#      compress=lz4
```
Comment (delgod) [inline deploy/cr.yaml:28]: could you please move it after `[mysqld]` section

Comment (tplavcic) [reply deploy/cr.yaml:31]: Moved.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | failed |
| big-data | failed |
| recreate | passed |
| init-deploy | passed |
| scaling | passed |
| limits | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | passed |
| monitoring | passed |
| self-healing-advanced | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| upgrade-consistency | passed |
| security-context | passed |
| affinity | passed |
| one-pod | passed |
| operator-self-healing | passed |
| auto-tuning | passed |
| scheduled-backup | passed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | failed |
| recreate | passed |
| init-deploy | failed |
| storage | passed |
| scaling | failed |
| scaling-proxysql | passed |
| self-healing | failed |
| restore-to-encrypted-cluster | failed |
| upgrade | failed |
| demand-backup | passed |
| limits | passed |
| self-healing-advanced | failed |
| monitoring | failed |
| operator-self-healing | passed |
| monitoring-2-0 | passed |
| scheduled-backup | failed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | failed |
| proxysql-sidecar-res-limits | passed |
| demand-backup-encrypted-with-tls | failed |
| upgrade-consistency | failed |

**Pull request merged**